docs/source/basics/chat_datasets.rst

```diff
@@ -0,0 +1,248 @@
+        {"role": "assistant", "content": "The answer is 42."},
+        {"role": "user", "content": "That's ridiculous"},
+        {"role": "assistant", "content": "Oh I know."},
+    ]
+
+This is more structured than freeform text association that models are typically pre-trained with,
+where they learn to simply predict the next token instead of responding accurately to the user.
+
+The primary entry point for fine-tuning with chat datasets in torchtune is the :func:`~torchtune.datasets.chat_dataset`
+builder. This lets you specify a local or Hugging Face dataset that follows the chat data format
+directly from the config and train your LLM on it.
+
+Example chat dataset
+--------------------
+
+.. code-block:: bash
+
+    head data/my_data.json -n 22
+    # [
+    #     {
+    #         "conversations": [
+    #             {
+    #                 "from": "human",
+    #                 "value": "What is the answer to life?"
+    #             },
+    #             {
+    #                 "from": "gpt",
+    #                 "value": "The answer is 42."
+    #             },
+    #             {
+    #                 "from": "human",
+    #                 "value": "That's ridiculous"
+    #             },
+    #             {
+    #                 "from": "gpt",
+    #                 "value": "Oh I know."
+    #             }
+    #         ]
+    #     }
+    # ]
+
+.. code-block:: python
+
+    from torchtune.models.mistral import mistral_tokenizer
+    from torchtune.datasets import chat_dataset
+
+    m_tokenizer = mistral_tokenizer(
+        path="/tmp/Mistral-7B-v0.1/tokenizer.model",
+        prompt_template="torchtune.models.mistral.MistralChatTemplate",
+        max_seq_len=8192,
+    )
+    ds = chat_dataset(
+        tokenizer=m_tokenizer,
+        source="json",
+        data_files="data/my_data.json",
+        split="train",
+        conversation_column="conversations",
+        conversation_style="sharegpt",
+        # By default, user prompt is ignored in loss. Set to True to include it
+        train_on_input=True,
+    )
+    tokenized_dict = ds[0]
+    tokens, labels = tokenized_dict["tokens"], tokenized_dict["labels"]
+    print(m_tokenizer.decode(tokens))
+    # [INST] What is the answer to life?  [/INST] The answer is 42. [INST] That's ridiculous  [/INST] Oh I know.
+    print(labels)
+    # [1, 733, 16289, 28793, 1824, 349, 272, 4372, ...]
+
+.. code-block:: yaml
+
+    # In config
+    tokenizer:
+      _component_: torchtune.models.mistral.mistral_tokenizer
+      path: /tmp/Mistral-7B-v0.1/tokenizer.model
+      prompt_template: torchtune.models.mistral.MistralChatTemplate
+      max_seq_len: 8192
+
+    dataset:
+      _component_: torchtune.datasets.chat_dataset
+      source: json
+      data_files: data/my_data.json
+      split: train
+      conversation_column: conversations
+      conversation_style: sharegpt
+      train_on_input: True
+      new_system_prompt: You are an AI assistant.
```

Comment:
this is missing from the above example? or no?

Reply:
ah, let me remove this. Mistral doesn't use system prompts